Merge 22.2 code freeze

mokagio opened the pull request:
- [x] Update internal Pods to stable version
- [x] New version header in `RELEASE-NOTE.txt` 
- [x] `Localizable.strings` updated 
- [x] `release_notes.txt` updated with notes from `RELEASE-NOTE.txt` for current version 
- [x] Version update in `.xcconfig`

mokagio committed:
Bump version number

mokagio committed:
Update draft release notes for 22.2.

mokagio committed:
Update draft release notes for 22.2.

mokagio committed:
Release Notes: add new section for next version (22.3)

mokagio committed:
Use production versions of internal pods for release 22.2

Notice the AppAuth patch version bump, which was due to running
`bundle exec pod update WordPressAuthenticator`.

mokagio committed:
Remove Jetpack-only header from raw release notes for Jetpack 22.2

mokagio committed:
Remove Jetpack-only entries from raw release notes for WordPress 22.2

mokagio committed:
Update strings for localization

mokagio commented on Podfile.lock:
```diff
@@ -4,11 +4,11 @@ PODS:
+    - AppAuth/Core (= 1.6.1)
+    - AppAuth/ExternalUserAgent (= 1.6.1)
+  - AppAuth/Core (1.6.1)
+  - AppAuth/ExternalUserAgent (1.6.1):
```
This changed because I run `bundle exec pod update WordPressAuthenticator`. Given it's a patch release, it seems appropriate to keep them.

mokagio commented on WordPress/Resources/en.lproj/Localizable.strings:
```diff
@@ -567,6 +567,9 @@
 "Add Contents to Post" = "Add Contents to Post";
 
+/* No comment provided by engineer. */
+"Add description" = "Add description";
```
There are quite a few strings without reverse-DNS key in this diff. All but [one](https://github.com/wordpress-mobile/WordPress-iOS/commit/1c40a9b2123af8d2bfd9490d66e233cc06c3737f#r109225555) come from libraries (Gutenberg, would be my guess). There's not much we can do about it right now, but it gave me the idea of possibly updating the automation that extracts the strings to generate the reverse DNS key based on the library name and the value. 

E.g.:

```
"pods.gutenberg.add_description" = "Add description";
```

fluiddot replied:
Yep, this string comes from the Jetpack's VideoPress block ([reference](https://github.com/Automattic/jetpack/blob/fd90454188d21d1bbcf16ff50dbdb74803729ccb/projects/packages/videopress/src/client/block-editor/blocks/video/components/details-panel/index.native.js#L29)).

mokagio replied:
Thanks for confirming @fluiddot 🙇‍♂️

spencertransier reviewed (approved):
Looks good to me! Just the conflict on `Podfile.lock`

mokagio committed:
Fix tests not compiling after WordPressKit 7.2.0 upgrade

mokagio committed:
Update WordPressKit to 8.0.0

This is necessary for the unit tests to pass after the 7.2.0 upgrade,
which had a bug.

staskus committed:
Fix Domains Service initializer crash when wordPressComRestApi is nil (#20559)

* Do not refresh domains if wordPressComRestApi is not initialized

* Only refresh domains when domains dashboard card feature is enabled

mokagio commented:
Closing in favor of #20560
